### PR TITLE
add tests in pkg/types package

### PR DESCRIPTION
**Pull request by yichengq**
for #1467

### CODE REVIEW

File: pkg/types/set_test.go [hunk 1]
```diff
@@ -164,3 +164,25 @@ func driveSetTests(t *testing.T, s Set) {
...
 	}
 }
+
+func TestUnsafeSetContainsAll(t *testing.T) {
+	vals := []string{"foo", "bar", "baz"}
+	s := NewUnsafeSet(vals...)
+
+	tests := []struct {
+		strs []string
+		wok  bool
```
Comment (xiang90) [inline pkg/types/set_test.go]: wcontain

File: pkg/types/urls_test.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+		{"mailto://127.0.0.1:4001"},
+		// not conform to host:port
+		{"http://127.0.0.1"},
+		// contain a path
+		{"http://127.0.0.1:4001/path"},
+	}
+	for i, tt := range tests {
+		_, err := NewURLs(tt)
+		if err == nil {
+			t.Errorf("#%d: want error, but nil", i)
```
Comment (xiang90) [inline pkg/types/urls_test.go]: but got

Comment (xiang90) [inline pkg/types/urls_test.go]: better: err = nil, want error

File: pkg/types/urls_test.go [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+		}
+	}
+}
+
+func TestNewURLs(t *testing.T) {
```
Comment (xiang90) [inline pkg/types/urls_test.go]: move this test to the top

File: pkg/types/urls_test.go [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+	}
+	for i, tt := range tests {
+		urls, _ := NewURLs(tt.strs)
+		if !reflect.DeepEqual(urls, tt.wurls) {
+			t.Errorf("#%d: urls = %+v, want %+v", i, urls, tt.wurls)
+		}
+	}
+}
+
+func mustNewURL(t *testing.T, s string) url.URL {
```
Comment (xiang90) [inline pkg/types/urls_test.go]: if i remember correctly, we already have two `mustNewURL` in the codebase. Can you please use the existing ones?

Comment (yichengq) [inline pkg/types/urls_test.go]: Yes!! it is in `pkg/testutil/`. will use.

Comment (yichengq) [inline pkg/types/urls_test.go]: hrm.. the one in testutil is implemented by this pkg. i will add a new one then.

Comment (yichengq) [inline pkg/types/urls_test.go]: obviously, it introduces a cyclone import then. pkg/testutil uses pkg/types/URLs, while pkg/types uses pkg/testutil to newURL. I think pkg/types is the root of the types, so i will keep mustNewURL here.

Comment (xiang90) [inline pkg/types/urls_test.go]: can you change mustNewURL to use the mustNewURL in this pkg?

Comment (yichengq) [inline pkg/types/urls_test.go]: It is weird to expose a test function(with t in arguments) in a normal package.

**Comment by xiang90** [comment]
lgtm

**Pull request merged**